app-frontend/src/app/components/histogram/nodeHistogram/nodeHistogram.controller.js

```diff
@@ -0,0 +1,236 @@
+            });
+            data = _.flatten(_.zip(data, offsetData));
+        }
+        if (this._masks.min || this.options.discrete) {
+            let last = _.last(data);
+            if (last.color === 'NODATA' || !this.options.discrete) {
+                data.splice(0, 0, {offset: data[0].offset, color: '#353C58'});
+                data.splice(0, 0, {offset: data[0].offset, color: '#353C58'});
+            } else {
+                data.splice(0, 0, {offset: data[0].offset, color: _.first(data.color)});
+                data.splice(0, 0, {offset: data[0].offset, color: last.color});
+            }
+        }
+        if (this._masks.max || this.options.discrete) {
+            let last = _.last(data);
+            if (last.color === 'NODATA' || !this.options.discrete) {
+                data.push({offset: _.last(data).offset, color: '#353C58'});
+                data.push({offset: _.last(data).offset, color: '#353C58'});
+            } else {
+                data.push({offset: _.last(data).offset, color: last.color});
+            }
+        }
+        linearGradient.attr('id', `line-gradient-${this.id}`)
+            .attr('gradientUnits', 'userSpaceOnUse')
+            .attr('x1', '0%').attr('y1', 0)
+            .attr('x2', '100%').attr('y2', 0)
+            .selectAll('stop')
+            .data(data)
+            .enter().append('stop')
+            .attr('offset', (d) => d.offset)
+            .attr('stop-color', (d) => d.color)
+            .attr('stop-opacity', (d) => Number.isFinite(d.opacity) ? d.opacity : 1.0);
+    }
+
+    setDefaults(options) {
+        this.options = Object.assign({
+            min: 0,
+            max: 255
+        }, options ? options : {});
+
+        this.histOptions = {
+            chart: {
+                type: 'lineChart',
+                showLegend: false,
+                showXAxis: false,
+                showYAxis: false,
+                yScale: d3.scale.log(),
+                margin: {
+                    top: 0,
+                    right: 0,
+                    bottom: 0,
+                    left: 0
+                },
+                height: 100,
+                xAxis: {
+                    showLabel: false
+                },
+                yAxis: {
+                    showLabel: false
+                },
+                tooltip: {
+                    enabled: false
+                },
+                dispatch: {
+                    renderEnd: () => {
+                        this.updateHistogramColors();
+                    }
+                }
+            }
+        };
+
+        if (!this._breakpoints) {
+            this._breakpoints = {
+                min: {
+                    value: this.options.min,
+                    checked: false
+                },
+                max: {
+                    value: this.options.max,
+                    checked: false
+                }
+            };
+        }
+    }
+
+    processDataToPlot(data) {
```

Comment:
I'm not sure if it's here or somewhere else, but I ran into one situation where the histogram was misleading.
This histogram looks like there is a smooth line from 0 to 150, but the actual histogram has just two values in it.
![misleadinghist](https://user-images.githubusercontent.com/447977/30120565-38b6fe44-92f7-11e7-9db4-e023a6271960.png)